src/sentry/analytics/event.py

```diff
@@ -9,23 +9,43 @@
 
 
 class Attribute(object):
-    __slots__ = ['name', 'type', 'required']
-
     def __init__(self, name, type=six.text_type, required=True):
         self.name = name
         self.type = type
         self.required = required
 
+    def serialize(self, value):
+        if value is None:
+            return value
+        return self.type(value)
+
 
 class Map(Attribute):
-    def __init__(self, name, attributes, required=True):
+    attributes = ()
+
+    def __init__(self, name, attributes=None, required=True):
         self.name = name
         self.required = required
-        self.attributes = attributes
+        if attributes:
```

Comment:
might not need this, but i was trying to move in the direction of having Event inherit from Map so I could just (more easily) reuse what is more or less duplicated code